### PR TITLE
Replace `xsalsa20poly1305` with `crypto_secretbox`

### PR DESCRIPTION
**Pull request by Sebbl0508**
`xsalsa20poly1305` is listed as deprecated

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ license = "ISC"
 name = "songbird"
 readme = "README.md"
 repository = "https://github.com/serenity-rs/songbird.git"
-version = "0.3.2"
+version = "0.3.3"
```
Comment (FelixMcFelix) [inline Cargo.toml]: Please don't change this: we'll handle this separately when we make a release at some later time.

File: Cargo.toml [hunk 2]
```diff
@@ -109,9 +109,9 @@ optional = true
 version = "0.8"
 features = ["v4"]
 
-[dependencies.xsalsa20poly1305]
+[dependencies.crypto_secretbox]
```
Comment (FelixMcFelix) [inline Cargo.toml]: Can you move the `[dependencies.crypto_secretbox]` block to its correct alphabetical place? I.e., between byteorder and dashmap.

File: Cargo.toml [hunk 3]
```diff
@@ -161,7 +161,7 @@ driver-core = [
     "typemap_rev",
     "url",
     "uuid",
-    "xsalsa20poly1305",
+    "crypto_secretbox",
```
Comment (FelixMcFelix) [inline Cargo.toml]: Ditto on alphabetical order.

File: src/driver/crypto.rs [hunk 4]
```diff
@@ -241,8 +244,10 @@ impl CryptoState {
 #[cfg(test)]
 mod test {
     use super::*;
+    use crypto_secretbox::{KeyInit, SecretBox};
     use discortp::rtp::MutableRtpPacket;
-    use xsalsa20poly1305::{aead::NewAead, KEY_SIZE, TAG_SIZE};
+    pub const KEY_SIZE: usize = SecretBox::<()>::KEY_SIZE;
+    pub const NONCE_SIZE: usize = SecretBox::<()>::KEY_SIZE;
```
Comment (FelixMcFelix) [inline src/driver/crypto.rs]: I don't think this line is necessary, and is also incorrect (`NONCE` != `KEY`).

File: src/driver/tasks/mixer.rs [hunk 5]
```diff
@@ -20,7 +21,8 @@ use rand::random;
 use std::{convert::TryInto, time::Instant};
 use tokio::runtime::Handle;
 use tracing::{debug, error, instrument};
-use xsalsa20poly1305::TAG_SIZE;
+
+const TAG_SIZE: usize = SecretBox::<()>::TAG_SIZE;
```
Comment (FelixMcFelix) [inline src/driver/tasks/mixer.rs]: Why not `use crate::driver::crypto::TAG_SIZE`?

**Review by FelixMcFelix — CHANGES_REQUESTED** [review]
Thanks, I wasn't aware that the library had been deprecated. There are some minor changes I'd like to ask for, otherwise I can see it's mostly a drop-in replacement.

**Comment by Sebbl0508** [comment]
I've adjusted the code. hope it looks good now

**Review by FelixMcFelix — APPROVED** [review]
Great, looks good to me! Thank you. :)

**Pull request merged**